creator-node/src/services/stateMachineManager/stateMonitoring/monitorState.jobProcessor.ts

```diff
@@ -181,15 +183,20 @@ async function monitorState({
 
     // Retrieve success metrics for all users syncing to their secondaries
     try {
-      userSecondarySyncMetricsMap =
-        await computeUserSecondarySyncSuccessRatesMap(users)
+      await secondarySyncHealthTracker.computeIfWalletOnSecondaryShouldContinueActions(
+        users.map((user) => ({
+          wallet: user.wallet,
+          secondary1: user.secondary1,
+          secondary2: user.secondary2
+        }))
+      )
       _addToDecisionTree(
         decisionTree,
         'computeUserSecondarySyncSuccessRatesMap Success',
         logger,
         {
-          userSecondarySyncMetricsMapLength: Object.keys(
-            userSecondarySyncMetricsMap
+          walletToSecondaryToShouldContinueActions: Object.keys(
+            secondarySyncHealthTracker.getWalletToSecondaryToShouldContinueAction()
```

Comment:
I see how you store it in the class instead of passing through the functions, but this won't work with Bull and cluster because one process could have one instance of the class, but another process with a different instance might handle the Bull job.

I think the functions way is actually better from a visibility and debugging standpoint anyway because you can see exactly what the function is taking in and putting out. but if you really don't want to pass it around you'd have to store it in yet another Redis key to avoid this concurrency issue

Reply:
refactored!